alibi_detect/od/sr.py

```diff
@@ -91,11 +218,36 @@ def saliency_map(self, X: np.ndarray) -> np.ndarray:
         -------
         Array with saliency map values.
         """
+        assert X.shape[0] > self.conv_amp.shape[0], "The length of the input signal should be greater " \
+                                                    "than the amplitude window"
```

Comment:
This sounds like we should raise an error instead if this is triggered by user error? Or is this something that should never really happen?

Reply:
I will raise an error